pkg/handler/loki.go

```diff
@@ -104,3 +90,26 @@ func getLokiError(resp []byte, code int) string {
 	}
 	return fmt.Sprintf("Error from Loki (code: %d): %s", code, message)
 }
+
+func executeFlowQuery(queryBuilder *loki.Query, lokiClient httpclient.HTTPClient) ([]byte, int, error) {
+	queryBuilder, err := queryBuilder.PrepareToSubmit()
+	if err != nil {
+		return nil, http.StatusBadRequest, err
+	}
+
+	flowsURL, err := queryBuilder.URLQuery()
+	if err != nil {
+		return nil, http.StatusBadRequest, err
+	}
+	hlog.Debugf("GetFlows URL: %s", flowsURL)
+
+	resp, code, err := lokiClient.Get(flowsURL)
+	if err != nil {
+		return nil, http.StatusServiceUnavailable, err
+	}
+	if code != http.StatusOK {
+		msg := getLokiError(resp, code)
+		return nil, http.StatusBadRequest, errors.New("Loki backend responded: " + msg)
```

Comment:
well, I think if we wanted to be really accurate we should handle the different possible error codes case by case. E.g. if Loki returns `404`, we shouldn't return `404` to the frontend, because it would mean that the frontend is trying to reach an non-existing page, which is not correct.
However I'm not sure about returning `BadRequest` , but don't know what is better here. In some situation that could be `InternalError`, in others that could be `ServiceUnavailable`, etc. There's no one-fits-all, but returning loki's code isn't good either, and I'm not sure it's really worth it to have a complex handling of loki's return codes, what really matters in the end is the error message, no?

Reply:
btw the loki return code is included in the message, so it's not lost